### ISSUE TITLE
`PolyhedralComplex` and `PolyhedralFan` constructor consistency

### ISSUE THREAD

**Issue by YueRen** [oscar-system/Oscar.jl#2315]
**Is your feature request related to a problem? Please describe.**
`PolyhedralComplex` and `PolyhedralFan` come with a comprehensive set of constructors which all work very nice.  But they are not very consistent with each other.  

For example, to construct a `PolyhedralComplex` from vertices and incidences, one must put incidences before vertices:
```
IM = IncidenceMatrix([[1,2,3],[1,3,4]]);
vr = [0 0; 1 0; 1 1; 0 1];
PC = PolyhedralComplex(IM, vr)
```
But to construct a `PolyhedralFan` from rays and incidences, one must put incidences after rays:
```
R = [1 0; 1 1; 0 1; -1 0; 0 -1];
IM = IncidenceMatrix([[1,2],[2,3],[3,4],[4,5],[1,5]]);
PF = PolyhedralFan(R,IM)
```

Similarly, though to a lesser extend:  There are two ways to construct a `Polyhedron`: 
1. via inequalities and equations: `Polyhedron(([-1 0; 1 0], [0,1]), ([0 1], [0]))`
2. via points, lineality, and rays: `convex_hull([0 0], [0 1], [1 0])`
In both cases, the pieces of information have to be passed as separate arguments of the constructor.

However, `PolyhedralComplex` requires vertices and rays in a single list, together with a vector that clarifies which entries are which:
```
VR = [0 0 0; 1 0 0; 0 1 0; -1 0 0];
IM = IncidenceMatrix([[1,2,3],[1,3,4]]);
far_vertices = [2,3,4];
PC = PolyhedralComplex(IM, VR, far_vertices)
```

**Describe the solution you'd like**
One suggestion could be
1. Change the constructor for `PolyhedralComplex` to
```
PolyhedralComplex{T}((vertices, vertex_incidences), (rays, ray_incidences), lineality)
```
In other words, pass vertices and rays as two tuples to `PolyhedralComplex` similar to how inequalities and equations are passed as two tuples to `Polyhedron`.

2. Put the (optional) lineality information in the constructor for `PolyhedralFan` last.

**Comment by fingolfin** [comment]
The argument order for `polyhedral_fan` was tweaked at some point; in the example here it is now `polyhedral_fan(IM,R)` which I think matches what @YueRen proposed.

**Comment by lkastner** [comment]
The argument order was fixed, so now we have:
```
julia> R = [1 0; 1 1; 0 1; -1 0; 0 -1];

julia> IM = IncidenceMatrix([[1,2],[2,3],[3,4],[4,5],[1,5]]);

julia> PF = PolyhedralFan(R,IM)
ERROR: MethodError: no method matching PolyhedralFan(::Matrix{Int64}, ::Polymake.LibPolymake.IncidenceMatrixAllocated{Polymake.LibPolymake.NonSymmetric})
Stacktrace:
 [1] top-level scope
   @ REPL[10]:1

julia> PF = polyhedral_fan(R,IM)
ERROR: MethodError: no method matching polyhedral_fan(::Matrix{Int64}, ::Polymake.LibPolymake.IncidenceMatrixAllocated{Polymake.LibPolymake.NonSymmetric})

Closest candidates are:
  polyhedral_fan(::IncidenceMatrix, ::Union{MatElem, AbstractVecOrMat}, ::Union{Nothing, MatElem, AbstractVecOrMat}; non_redundant)
   @ Oscar ~/oscars/master/src/PolyhedralGeometry/PolyhedralFan/constructors.jl:103
  polyhedral_fan(::IncidenceMatrix, ::Union{MatElem, AbstractVecOrMat}; non_redundant)
   @ Oscar ~/oscars/master/src/PolyhedralGeometry/PolyhedralFan/constructors.jl:109
  polyhedral_fan(::Union{Field, Type{<:Union{Float64, FieldElem}}}, ::IncidenceMatrix, ::Union{MatElem, AbstractVecOrMat}, ::Union{Nothing, MatElem, AbstractVecOrMat}; non_redundant)
   @ Oscar ~/oscars/master/src/PolyhedralGeometry/PolyhedralFan/constructors.jl:67
  ...

Stacktrace:
 [1] top-level scope
   @ REPL[11]:1

julia> PF = polyhedral_fan(IM, R)
Polyhedral fan in ambient dimension 2
```
Also Alexej fixed the rest in:
```
lars@nomurai:~/oscars/master/src/PolyhedralGeometry/PolyhedralComplex$ git show 0887b942097
commit 0887b9420972382e772b50a9e017a21571793d16
Author: Alexej Jordan <jordan@math.tu-berlin.de>
Date:   Wed Jun 28 19:41:52 2023 +0200

    added polyhedral_complex construction method with signature analoguous to the rest of the polyhedral objects
    [skip ci]
```

**Comment by YueRen** [comment]
@lkastner Thanks for the fix, Lars!
